### PR TITLE
Auto branch update

### PR DESCRIPTION
**Pull request by apoorvsadana**

### CODE REVIEW

File: src/utils/madara.rs [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
-    let repo_url = format!("{}/{}/{}", GITHUB_BASE_URL, KARNOT_REPO_ORG, MADARA_REPO_NAME);
+pub fn clone_madara_and_build_repo(config: &AppChainConfig) -> Result<(), MadaraError> {
+    let repo_url = format!("{}/{}/{}", GITHUB_BASE_URL, MADARA_REPO_ORG, MADARA_REPO_NAME);
     let madara_path = get_madara_home()?.join("madara");
 
-    match git_clone(&repo_url, &madara_path, Some(BRANCH_NAME)) {
+    let madara_version = match config.madara_version.as_str() {
+        // there was a bug initially where the wrong commit version was being set
+        // in the config
+        "523ceedc8afe7a4f58abfdbb915aff3c36e817fe" => "5de416aeb2d9e4297e58f7f2dff99aeae521855e",
```
Comment (anshalshukla) [inline src/utils/madara.rs:18]: Why are we replacing this, looks weird.

File: src/utils/github.rs [hunk 2]
```diff
@@ -14,22 +14,19 @@ struct Commit {
...
 
-    return match response {
-        Ok(response) => match response.json::<Vec<Commit>>().await {
-            Ok(commits) => match commits.first() {
-                Some(latest_commit) => Ok(latest_commit.sha.clone()),
-                None => Err(GithubError::NoCommitsFound),
-            },
+    match response {
+        Ok(response) => match response.json::<Commit>().await {
+            Ok(latest_commit) => Ok(latest_commit.sha.clone()),
```
Comment (anshalshukla) [inline src/utils/github.rs:25]: we don't need to call `first()`?

**Review by anshalshukla — APPROVED** [review]

File: src/utils/github.rs (hunk 2 above)
Comment (apoorvsadana) [reply src/utils/github.rs:25]: no, when we add the branch to the api it automatically gives the latest commit

File: src/utils/madara.rs (hunk 1 above)
Comment (apoorvsadana) [reply src/utils/madara.rs:18]: There was a bug in the code where we were setting Madara version to the latest commit on the `keep-starknet-strange` repo. However, we are actually cloning our fork of the repo, so there was a mismatch. All older deployments have the wrong commit in their toml so manually changing it here.

**Pull request merged**